docs/resources/swr_domain.md

```diff
@@ -0,0 +1,64 @@
+---
+subcategory: "Software Repository for Container (SWR)"
+---
+
+# huaweicloud_swr_domain
+
+Manages a SWR domain resource within HuaweiCloud.
+
+## Example Usage
+
+```hcl
+variable "organization_name" {} 
+variable "repository_name" {}
+variable "access_domain" {}
+
+resource "huaweicloud_swr_domain" "test" {
+  organization  = var.organization_name
+  repository    = var.repository_name
+  access_domain = var.access_domain
+  permission    = "read"
+  deadline      = "forever"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
+  provider-level region will be used. Changing this creates a new resource.
+
+* `organization` - (Required, String, ForceNew) Specifies the name of the organization (namespace) the domain belongs.
+  Changing this creates a new resource.
+
+* `repository` - (Required, String, ForceNew) Specifies the name of the repository the domain belongs.
+  Changing this creates a new resource.
+
+* `access_domain` - (Required, String, ForceNew) Specifies the name of the domain for image sharing.
+  Changing this creates a new resource
+  -> **NOTE:** `access_domain` should be an existing Huaweicloud domain.
+
+* `permission` - (Required, String) Specifies the permission to be granted. Currently, only the **read** permission is supported.
+
+* `deadline` - (Required, String) Specifies the end date of image sharing (UTC). When the value is set to forever,
```

Comment:
UTC is a time format, I can add an example to make it more clear.
I think we should keep the description of `forever` . Because it's a higher privilege the other deadline, and it's not good to make it as a default. We don't want to let our user give other domain permanent privilege by mistake.